### ISSUE TITLE
State listeners for an attribute fire even though the attribute's value hasn't changed

### ISSUE THREAD

**Issue by efficiosoft** [home-assistant/appdaemon#512]
Hi,

Observed on AppDaemon 3.0.2.

A state listener registered with

    self.listen_state(callback, entity, attribute="operation_mode")

fires when just the ``temperature`` attribute has changed and ``operation_mode`` is still the same. The values passed to the callback for ``old`` and ``new`` are then identical.

My assumption is that it shouldn't fire at all unless ``operation_mode`` changes, right?

Best regards
Robert

**Comment by Odianosen25** [comment]
Yeah issue needs looking into, as when another attribute is changed on the entity, the attribute being listened to is called even though it didn't change or was not affected.

In a busy thread, this easily leads to thread starvation due to constant calling of the attribute, which sometimes seem it leads to some form of loop for a while.

Regards

**Comment by raffidil** [comment]
I have the exact same issue. Looking forward to hearing from it.

**Comment by Odianosen25** [comment]
This has been fixed in latest dev, so closing this issue

**Comment by ZzetT** [comment]
Is this really fixed?

I'm using AppDaemon 4.0.2 so this fix should be already included.

However, if try the following:
```
class HelloWorld(hass.Hass):

  def initialize(self):
    self.log("Hello from AppDaemon")
    self.listen_state(self.motion, "binary_sensor.movement1", new = "on")

     
     
  def motion(self, entity, attribute, old, new, kwargs):
    self.log("Motion detected")
```

and trigger the motion, I see the following in the logs (the sensor stays on for two minutes):

```
2019-08-11 22:34:32.533551 INFO hello_world: Motion detected
2019-08-11 22:36:02.456153 INFO hello_world: Motion detected
```

As soon as I add
```
    if new == old:
      return
```
I see it only once. Also if I use an HA automation, I only see it once in the log so the sensor shouldn't be the problem here.

**Comment by Odianosen25** [comment]
Hello @ZzetT,

Yes this has been fixed and you can't be using a `4.0.2` as AD 4.0 hasn't been released, and its still in dev as `4.0.0b1`. So please confirm the version you using, as if you still in `3.` its not fixed there.

Kind regards

**Comment by ZzetT** [comment]
@Odianosen25  Sorry for the confusion. I'm using this HA addon https://github.com/hassio-addons/addon-appdaemon3 which is version 4.0.2 but uses version 3.0.5 of appdaemon.

There I can observe the above mentioned problem.

**Comment by Odianosen25** [comment]
Hello @ZzetT,

If that is the case, then no its not fixed in 3.0.5. You will have to wait for AD 4.0 to be released I a afraid. In there its definitely fixed.

Kind regards
